stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -260,11 +303,21 @@ public struct Unsafe${Mutable}BufferPointer<Element>
         base: self, bounds: bounds)
     }
 %  if Mutable:
-    set {
+    nonmutating set {
       _debugPrecondition(bounds.lowerBound >= startIndex)
       _debugPrecondition(bounds.upperBound <= endIndex)
+      _debugPrecondition(bounds.count == newValue.count)
+
       // FIXME: swift-3-indexing-model: tests.
-      _writeBackMutableSlice(&self, bounds: bounds, slice: newValue)
+      if newValue.count > 0 {
```

Comment:
This should be `!newValue.isEmpty`. I realize it's not a big deal in this case since we're talking about integer-indexed collections here but we should never use `count` to check for emptiness.

Reply:
Feel free to change `UnsafeMutableRawBufferPointer` implementation as well.